recipes/picojson/all/test_package/conanfile.py

```diff
@@ -0,0 +1,22 @@
+from conans import ConanFile, CMake, tools
+
+import os
+
+class PicoJSONTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake_find_package"
+
+    def build(self):
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
+        cmake.configure()
+        self._is_multi_configuration = cmake.is_multi_configuration
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = "example"
+            if self._is_multi_configuration:
```

Comment:
Why do you need it? It's not a regular flow.

Reply:
Because in the test package I use cmake_find_package generator, not cmake. Is cmake_find_package a bad practice in test packages?

Reply:
No, you can use cmake_find_package, no problem. But I think you are looking for `cmake-find-package-multi` generator, because `cmake_find_package` does not use build type folder.

Reply:
Now I see, you using only `cmake_find_package`, that's why you don't have the `bin` folder. Okay, my mistake.

Reply:
I tested this recipe locally with [Windows, VS 15] configuration and had no problems with finding example.exe from the test package. But the recipe failed with [Linux, gcc] most probably because CMake generator for the former case is multi-configuration and single-configuration for the latter, so I added _is_multi_configuration. The output is placed in `<config>` subfolder due to standard CMake behavior for multi-config. But this behavior is changed by conan_basic_setup() which is used along with conan cmake generator. `conan_basic_setup()` invokes [conan_output_dirs_setup()](https://github.com/conan-io/conan/blob/1.20.4/conans/client/generators/cmake_common.py#L323) (if `NO_OUTPUT_DIRS` is not provided) that is why many recipes rely on the "bin" folder to look for an output. But since I don't use cmake generator (I prefer non-intrusive approach - not to add `conan_basic_setup()` to CMakeLists) I adjust the output folder in conan test() method. But I am not sure that I need cmake_find_package_multi generator here..

Reply:
Oops, seems my reply is a bit late, but i will leave it here :)